### PR TITLE
Reland "Implementing TextScaler for nonlinear text scaling (#42062)"

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
The original PR crashes because of a JNI signature mismatch (`(FJ)F` -> `(FI)F`). Update the signature in https://github.com/flutter/engine/commit/a415da561964e101392504845f70887a46a31161

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I signed the [CLA].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/android/test/io/flutter/embedding/engine/systemchannels/SettingsChannelTest.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    executor.onAttachedToJNI();
+    final SettingsChannel settingsChannel = new SettingsChannel(executor);
+
+    final ArgumentCaptor<ByteBuffer> messageCaptor = ArgumentCaptor.forClass(ByteBuffer.class);
+
+    settingsChannel.startMessage().setDisplayMetrics(mock(DisplayMetrics.class)).send();
+
+    verify(executor).send(eq("flutter/settings"), messageCaptor.capture(), isNull());
+  }
+
```
Comment (reidbaker) [inline shell/platform/android/test/io/flutter/embedding/engine/systemchannels/SettingsChannelTest.java:37]: Can you add a test with minSdk = 33 that will eventually exercise the api 34 code or at minimum a todo with link to a bug in this file to add a tests once api 34 is available against roboletric.

Comment (LongCatIsLooong) [reply shell/platform/android/test/io/flutter/embedding/engine/systemchannels/SettingsChannelTest.java:37]: Added a TODO that points to https://github.com/flutter/flutter/issues/128825

**Review by reidbaker — APPROVED** [review]
